install.sh

```diff
@@ -1,62 +1,167 @@
-	if [[ -e /etc/debian_version ]]; then
-		sudo apt purge jq
-	elif [[ -e /etc/fedora-release ]]; then
-		sudo dnf remove jq
-	elif [[ -e /etc/centos-release ]]; then
-		sudo yum remove jq
-	elif [[ -e /etc/arch-release ]]; then
-		sudo pacman -Rs jq
-	fi
-else
-	echo "Did not uninstall jq as it was preinstalled before running this script."
+}
+
+install_latest_release () {
+    echo "We are now going to download the code for YT-Spammer-Purge."
+    sleep 1
+    git clone https://github.com/ThioJoe/YT-Spammer-Purge
+    cd YT-Spammer-Purge
+    git checkout -q -m $(git describe --abbrev=0 --tags)
+    # Go to latest release
+}
+
+install_MAIN () {
+    clear
+    echo "We will now install YT-Spammer-Purge. If this is not what you intend to do, exit within 5 seconds..."
+    sleep 5
+    echo "Installing."
+    # Check what OS we're running on
+
+    # Check if we already installed the requirements -- git is needed to check for which mode to run.
+    [[ $REQUIREMENTS_INSTALLED -eq 0 ]] && install_os_requirements
+
+    echo "--------------------------"
+
+    install_latest_release
+
+    # Since we've gotten python3 installed:
+
+    echo "--------------------------"
+    install_python_requirements
+
+    # Done!
+
+    printf "Dependencies and Program installed into .\YT-Spammer-Purge!\nNow follow these instructions to get a client_secrets.json file!\nhttps://github.com/ThioJoe/YT-Spammer-Purge/wiki/Instructions:-Obtaining-an-API-Key\n\nYou may run this script again inside your installation to update.\n"
+    exit 0
+}
+
+update () {
+    clear
+    echo "We will now update YT-Spammer-Purge. If this is not what you intend to do, exit within 5 seconds..."
+    sleep 5
+    echo "Continuing."
+    echo "Current version is $(git describe --abbrev=0 --tags)"
+    echo "Updating..."
+
+
+    git fetch origin
+    echo "Latest version is $(git describe origin --abbrev=0 --tags)"
+    echo "Updating to this version."
+    git checkout -q -m $(git describe origin --abbrev=0 --tags)
+
+
+    echo "--------------------------"
+    echo "Updated!"
+    echo "Report any bugs to TJoe.io/bug-report"
+    exit 0
+}
+
+check_git_missing () {
+    [[ $(git remote get-url origin) == *"YT-Spammer-Purge"* ]] && return 0
+    # If this is a valid YT-Spammer-Purge install, return 0
+    clear
+    echo "It looks like you downloaded a .zip of YT-Spammer-Purge"
+    echo "Automated updates do not work on these versions, but you may download the latest version of YT-Spammer-Purge using this script."
+    echo "If you choose to re-download the latest verion of YT-Spammer-Purge using this script, automated updates will be re-enabled."
+    echo "The latest YT-Spammer-Purge with automated updates will be downloaded to a sub-directory of the same name."
+    echo "If you would not like to install YT-Spammer-Purge using this script, close this script within the next 15 seconds..."
+    sleep 15
+    install_MAIN
+    exit 0
+}
+
+if ! command -v git &> /dev/null
```

Comment:
You are only checking if git is present/missing. This check is not sufficient, because it can happen that git is available but we are missing other dependencies. That would break the install script. Either check for all dependencies and install accordingly or install them everytime no matter what.

Reply:
Git is the only pre-requisite for this script, as this script uses git to manage the version (update) and install the program, so I check for it before any git commands are ran 👍
During installation, all other requirements are installed as well.

I'll add a check for python3 as well, and run the `install_os_requirements` function on that case
I think I may be able to check for the other requirements as well.